### DISCUSSION TITLE
using a binary cache files provider with docker compose container

### DISCUSSION THREAD

**Discussion by willie** [microsoft/vcpkg#31567] [Q&A] [answered]
For local development, I'm trying to use a binary cache in between docker builds on my local machine. I'm having a hard time with the files provider. I have the following in my docker-compose.yml:

```
    volumes:
      - ~/.cache/vcpkg/archives:/root/.cache/vcpkg/archives
    environment:
      - VCPKG_BINARY_CACHE="files,/root/.cache/vcpkg/archives,<rw>"
```

But the cache isn't being used, so I'm at a bit of a loss. Any suggestions?

**Comment by willie** [comment]
After reading some discussions, I tried this:
```
VCPKG_BINARY_CACHE=clear;files,/root/.cache/vcpkg/archives,readwrite
```
But it didn't work either. Honestly, I don't understand what the syntax should be without examples based on the docs.

**Comment by willie** [comment]
Ugh, I think I just realized the issue: docker build doesn't have access to volumes. 🤦 I'm going to leave this open if there are workarounds.

**Comment by willie** [comment]
Actually, I can't get any binary caching to really work. I tried GCS with:

```
VCPKG_BINARY_CACHE="clear;x-gcs,gs://BUCKETNAME/,readwrite"
```

After confirming that `gsutil` is working, so I'm at a loss. Is there a debug that can tell me if it's being configured?

**Comment by autoantwort** [comment, marked as answer]
Maybe try setting the right variable ;) The variable is called [`VCPKG_BINARY_SOURCES`](https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_binary_sources) and not `VCPKG_BINARY_CACHE` :)

**Reply by willie** [reply]
🤦‍♂️ 

Thank you so much for seeing where I made that mistake. I'm not sure how I mixed that up. Thanks again!
